src/test/java/de/tum/in/www1/artemis/service/connectors/LtiServiceTest.java

```diff
@@ -0,0 +1,286 @@
+    }
+
+    private void onSuccessfulAuthenticationAssertions(User user, LtiUserId ltiUserId) {
+        assertThat(user.getGroups().contains(courseStudentGroupName));
+        assertThat(ltiUserId.getLtiUserId().equals("ff30145d6884eeb2c1cef50298939383"));
+        assertThat(ltiOutcomeUrl.getUrl().equals("some.outcome.service.url.com"));
+        assertThat(ltiOutcomeUrl.getSourcedId().equals("someResultSourceId"));
+        verify(userService, times(1)).save(user);
+        verify(artemisAuthenticationProvider, times(1)).addUserToGroup(user, courseStudentGroupName);
+        verify(ltiOutcomeUrlRepository, times(1)).save(ltiOutcomeUrl);
+    }
+
+    @Test
+    public void verifyRequest_oauthSecretNotSpecified() {
+        ReflectionTestUtils.setField(ltiService, "OAUTH_SECRET", Optional.empty());
+        HttpServletRequest request = mock(HttpServletRequest.class);
+
+        String message = ltiService.verifyRequest(request);
+
+        assertThat(message.equals("verifyRequest for LTI is not supported on this Artemis instance, artemis.lti.oauth-secret was not specified in the yml configuration"));
+    }
+
+    @Test
+    public void verifyRequest_unsuccessfulVerification() {
+        ReflectionTestUtils.setField(ltiService, "OAUTH_SECRET", Optional.of("secret"));
+        String url = "http://some.url.com";
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        when(request.getHeader(anyString())).thenReturn(null);
+        when(request.getRequestURL()).thenReturn(new StringBuffer(url));
+        when(request.getMethod()).thenReturn("GET");
+        when(request.getHeaderNames()).thenReturn(Collections.emptyEnumeration());
+        when(request.getParameterNames()).thenReturn(Collections.emptyEnumeration());
+
+        String message = ltiService.verifyRequest(request);
+
+        assertThat(message.equals("LTI signature verification failed with message: Failed to validate: parameter_absent; error: bad_request, launch result: null"));
+    }
+
+    @Test
+    public void onNewResult() {
+        ReflectionTestUtils.setField(ltiService, "OAUTH_KEY", Optional.of("oauthKey"));
+        ReflectionTestUtils.setField(ltiService, "OAUTH_SECRET", Optional.of("oauthSecret"));
+
+        StudentParticipation participation = new StudentParticipation();
+        User user = new User();
+        participation.setParticipant(user);
+        participation.setExercise(exercise);
+        participation.setId(27L);
+        Result result = new Result();
+        result.setScore(3L);
+        LtiOutcomeUrl ltiOutcomeUrl = new LtiOutcomeUrl();
+        ltiOutcomeUrl.setUrl("https://some.url.com/");
+        ltiOutcomeUrl.setSourcedId("sourceId");
+        ltiOutcomeUrl.setExercise(exercise);
+        ltiOutcomeUrl.setUser(user);
+
+        when(ltiOutcomeUrlRepository.findByUserAndExercise(user, exercise)).thenReturn(Optional.of(ltiOutcomeUrl));
```

Comment:
this line is used more than once.  
Try to setup the mocking in the init() function or another setup() function.